fastly/erl_test.go

```diff
@@ -0,0 +1,243 @@
+package fastly
+
+import (
+	"testing"
+)
+
+func TestClient_ERL(t *testing.T) {
+	t.Parallel()
+
+	fixtureBase := "erls/"
+	testVersion := createTestVersion(t, fixtureBase+"version", testServiceID)
+
+	// Create
+	var err error
+	var e *ERL
+	var saveID string
+	record(t, fixtureBase+"create", func(c *Client) {
+		e, err = c.CreateERL(&CreateERLInput{
+			ServiceID:          testServiceID,
+			ServiceVersion:     testVersion.Number,
+			Name:               "test_erl",
+			Action:             ERLActionResponse,
+			ClientKey:          []string{"req.http.Fastly-Client-IP"},
+			HttpMethods:        []ERLHttpMethodsEnum{ERLGet, ERLPost},
+			PenaltyBoxDuration: 30,
+			Response: &ERLResponseType{
+				ERLStatus:      429,
+				ERLContentType: "application/json",
+				ERLContent:     "Too many requests"},
+			RpsLimit:   20,
+			WindowSize: 10,
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	saveID = e.ID
+
+	// Ensure deleted
+	defer func() {
+		record(t, fixtureBase+"cleanup", func(c *Client) {
+			c.DeleteERL(&DeleteERLInput{
+				ServiceID:      testServiceID,
+				ServiceVersion: testVersion.Number,
+				ERLID:          e.ID,
+			})
+		})
+	}()
+
+	if e.Name != "test_erl" {
+		t.Errorf("bad name: %q", e.Name)
+	}
```

Comment:
Might be worth validating more fields.

Reply:
Added an additional param and can add more as needed